nmtwizard/preprocess/tu.py

```diff
@@ -210,22 +220,34 @@ def __init__(self,
                  alignment=None,
                  source_tokenizer=None,
                  target_tokenizer=None):
-        self.__source = [TranslationSide(source, tokenizer=source_tokenizer)]
+        self.__source = {"main": TranslationSide(source, "source", tokenizer=source_tokenizer)}
```

Comment:
Do we need a dictionary here? Can't we just consider the first index as "main" and others as "extra"?

Reply:
In source we probably could. But in target, as we discussed, in inference we wouldn't know if the first target came from a translation prefix or it's an extra target. We could use a list with names if you prefer. I thought dictionary will ensure unique names.